task.sql

```diff
@@ -1 +1,38 @@
-# Write your SQL code for the database creation here. Good luck! 
+CREATE DATABASE ShopDB;
+
+USE ShopDB;
+
+CREATE TABLE Products (
+    ID INT AUTO_INCREMENT,
+    ProductName VARCHAR(50),
+    Description TEXT,
+    Price INT,
+    WarehouseAmount INT,
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE Customers (
+    ID INT AUTO_INCREMENT,
+    FirstName VARCHAR(50),
+    LastName VARCHAR(50),
+    Email VARCHAR(50),
+    Address VARCHAR(255),
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE Orders (
+    OrdersID INT AUTO_INCREMENT,
+    CustomerID INT,
+    Date DATE,
+    FOREIGN KEY (CustomerID) REFERENCES Customers(ID) ON DELETE SET NULL,
```

Comment:
The foreign key constraint on `CustomerID` in the `Orders` table uses `ON DELETE SET NULL`. However, `CustomerID` is an `INT` and cannot be set to `NULL` unless it is explicitly defined as `NULL`. Consider changing `CustomerID INT` to `CustomerID INT NULL` to allow this behavior.